quinn-udp/Cargo.toml

```diff
@@ -30,8 +30,13 @@ once_cell = { workspace = true }
 windows-sys = { workspace = true }
 
 [dev-dependencies]
-criterion = "0.5"
+criterion = { version = "0.5", default-features = false, features = ["async_tokio"] }
+tokio = { workspace = true, features = ["rt", "rt-multi-thread", "net"] }
```

Comment:
On Linux blocking `recvmmsg` calls will wait for all `iovec`s to be filled. 

>  A blocking recvmmsg() call blocks until vlen messages have been
       received or until the timeout expires.  A nonblocking call reads
       as many messages as are available (up to the limit specified by
       vlen) and returns immediately.

https://man7.org/linux/man-pages/man2/recvmmsg.2.html

This is e.g. problematic when combined with a GSO send with less segments than `iovec`s in its `recvmmsg` counterpart.

Thus this pull request changes the benchmark to use non-blocking instead of blocking `recvmmsg`. To do so, it uses `tokio`.

See also discussion in https://github.com/quinn-rs/quinn/pull/1993#discussion_r1786705730.

Reply:
Suggest you keep a separate commit that makes the change from sync to async.

Reply:
:+1: 5609a1d3